src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4592,6 +4592,40 @@ public override BoundNode VisitThrowExpression(BoundThrowExpression node)
             return result;
         }
 
+        public override BoundNode VisitYieldReturnStatement(BoundYieldReturnStatement node)
+        {
+            BoundExpression expr = node.Expression;
+            if (expr == null)
+            {
+                return null;
+            }
+            var method = (MethodSymbol)_member;
+            Debug.Assert(method.IsIterator);
+            VisitOptionalImplicitConversion(expr, getIteratorElementType(method.ReturnType.TypeSymbol), useLegacyWarnings: false, AssignmentKind.Return);
+            return null;
+
+            TypeSymbolWithAnnotations getIteratorElementType(TypeSymbol returnType)
```

Comment:
Looks like we have a helper GetIteratorElementTypeFromReturnType that already exists and is almost identical to this function. Can we share?

Reply:
I considered and tried it, but the code ended up worse in my opinion.
To make a static method, we need to pass two extra parameters (binder and compilation). Trying to only use the compilation and not the binder is possibly but was trouble.
Then the two methods also differ in their handling of errors. The original method uses a `GetSpecialType` that requires a diagnostic bag (I could pass a dummy one I suppose).

Let me know what you think.

Reply:
FWIW, here's what that looks like:
https://github.com/jcouv/roslyn/commit/e81acf586a91f7b970cb7155f5c69f1ab987c838